### PR TITLE
:books: Update documentation

### PR DESCRIPTION
**Pull request by elbeno**
Problem:
 - The `atomic` interface is not documented.
 - The injection mechanism for the `conc` interface is not documented.

Solution:
 - Improve the docs to cover the `atomic` interface and customization points in general.

### CODE REVIEW

File: docs/index.adoc [hunk 1]
```diff
@@ -13,6 +13,135 @@ This C++ baremetal concurrency library provides a low-level API for operations
...
+               std::memory_order mo = std::memory_order_seq_cst) -> T;
+----
+
+=== Customization of atomic operations
+
+By default, the atomic interface surfaced in the `atomic` namespace is
+implemented by `atomic::standard_policy`, a policy class that has static
+functions corresponding to each of the above functions. And the
+`standard_policy` functions call
+https://llvm.org/docs/Atomics.html#libcalls-atomic[the compiler intrinsics]
```
Comment (lukevalenty) [inline docs/index.adoc:71]: question: are these intrinsics the same as gcc's?

Comment (lukevalenty) [reply docs/index.adoc:71]: (I'm assuming "yes", if so, I think the documentation should point that out)

**Review by lukevalenty — APPROVED** [review]
awesome 😎

File: docs/index.adoc (hunk 1 above)
Comment (elbeno) [reply docs/index.adoc]: Yes. Added a note.

**Pull request merged**